CourseGrab/ViewControllers/Search/SearchTableViewCell.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  SearchTableViewCell.swift
+//  CourseGrab
+//
+//  Created by Reade Plunkett on 2/4/20.
+//  Copyright © 2020 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+class SearchTableViewCell: UITableViewCell {
+
+    private let arrowImageView = UIImageView()
+    private let container = UIView()
+    private let titleLabel = UILabel()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        // Setup appearance
+
+        selectionStyle = .none
+
+        //arrowImageView.image = UIImage(named: "logo_AppDev_black")
+        arrowImageView.backgroundColor = .black
+        arrowImageView.contentMode = .scaleAspectFit
+        container.addSubview(arrowImageView)
+
+        container.clipsToBounds = false
+        container.layer.cornerRadius = 5
+        container.layer.shadowColor = UIColor.black.cgColor
+        container.layer.shadowOpacity = 0.3
+        container.layer.shadowRadius = 2
+        container.layer.shadowOffset = .zero
+        container.backgroundColor = .white
+        contentView.addSubview(container)
+
+        titleLabel.font = ._16Semibold
+        titleLabel.numberOfLines = 0
+        container.addSubview(titleLabel)
+
+        // Setup constraints
+
+        arrowImageView.snp.makeConstraints { make in
+            make.size.equalTo(16)
+            make.centerY.equalTo(titleLabel.snp.centerY)
+            make.leading.equalTo(titleLabel.snp.trailing).inset(8)
+            make.trailing.equalToSuperview().inset(16)
+        }
+
+        container.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(6)
+            make.bottom.equalToSuperview().inset(6)
+            make.leading.equalTo(20)
+            make.trailing.equalToSuperview().inset(20)
+        }
+
+        titleLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(16)
+            make.trailing.equalTo(arrowImageView.snp.leading).inset(-8)
+            make.top.bottom.equalToSuperview().inset(16)
+        }
+    }
+
+    func configure(for section: Section) {
+        titleLabel.text = section.title
+
+        if section.isTracking {
+            // Setup appearance
+
+            let divider = UIView()
+            divider.backgroundColor = .courseGrabDivider
+            container.addSubview(divider)
+
+            let trackingButton = UIButton()
+            trackingButton.layer.cornerRadius = 2
+            trackingButton.layer.borderWidth = 1
+            trackingButton.layer.borderColor = UIColor.courseGrabRuby.cgColor
+            trackingButton.titleLabel?.font = ._12Semibold
+            container.addSubview(trackingButton)
```

Comment:
because `configure` is being used in `cellForItemAt`, that means this will be called multiple times for the same cell. in general, all the views should be created inside the `init` and inside the `configure` you can `make/remake` constraints as needed

Reply:
^ Absolutely address this!